flink-runtime/src/main/java/org/apache/flink/runtime/state/OperatorStateRestoreOperation.java

```diff
@@ -168,9 +170,25 @@ public Void restore() throws Exception {
                     }
                 }
 
+                // sort state by offsets
```

Comment:
How about describing the root cause here?
something like: 
>  sort state handles by offsets to avoid building `SnappyFramedInputStream` with EOF stream.

And IIUC,  we only need to sort the state handles when compression is enabled?

Reply:
Done.